rocks/RocksDatabase.java

```diff
@@ -239,10 +236,25 @@ private void mayInitRocksLogger() {
         }
     }
 
-    private void validateEncoding(RocksStorage.Schema schemaStorage) {
-        int encoding = schemaStorage.getEncoding();
-        if (encoding != ENCODING_VERSION) {
-            throw TypeDBException.of(INCOMPATIBLE_ENCODING, name(), encoding, ENCODING_VERSION);
+    private void initialiseEncodingVersion() {
+        try {
+            rocksSchema.put(rocksSchemaPartitionMgr.get(Storage.Key.Partition.DEFAULT),
+                    ENCODING_VERSION_KEY.bytes().getBytes(), ByteArray.encodeInt(ENCODING_VERSION).getBytes());
+        } catch (RocksDBException e) {
+            throw TypeDBException.of(e);
+        }
+    }
+
+    private void validateEncodingVersion() {
+        try {
+            byte[] encodingBytes = rocksSchema.get(rocksSchemaPartitionMgr.get(Storage.Key.Partition.DEFAULT),
+                    ENCODING_VERSION_KEY.bytes().getBytes());
+            int encoding = encodingBytes == null || encodingBytes.length == 0 ? 0 : ByteArray.of(encodingBytes).decodeInt();
+            if (encoding != ENCODING_VERSION) {
+                throw TypeDBException.of(INCOMPATIBLE_ENCODING, name(), encoding, ENCODING_VERSION);
+            }
+        } catch (RocksDBException e) {
+            throw TypeDBException.of(e);
```

Comment:
These method seems to have changed from how they were written before. Add a comment to explain what changed and why?